source/github-helpers/search-query.ts

```diff
@@ -1,17 +1,43 @@
 import {getUsername} from '.';
 
-type Source = HTMLAnchorElement | URL | URLSearchParams | Location;
+type Source = HTMLAnchorElement | URL | string | string[][] | Record<string, string> | URLSearchParams;
+
+const queryPartsRegExp = /(?:[^\s"]+|"[^"]*")+/g;
+function splitQueryString(query: string): string[] {
+	return query.match(queryPartsRegExp) ?? [];
+}
+
+// Remove all values from array exept the last occurence of one of the values.
+function deduplicate(array: any[], ...values: any[]): any[] {
+	let indexToKeep = -1;
+	for (const value of values) {
+		const lastIndex = array.lastIndexOf(value);
+		if (lastIndex > indexToKeep) {
+			indexToKeep = lastIndex;
+		}
+	}
+
+	if (indexToKeep === -1) {
+		return array;
+	}
+
+	return array.filter((value, index) => index >= indexToKeep || !values.includes(value));
```

Comment:
```suggestion
	let alreadyFound = false;
	const newArray = []
	for (const item of array.reverse()) {
		const found = values.includes(item);
		if (!alreadyFound || !found) {
			newArray.unshift(item)
		}
		
		alreadyFound = alreadyFound || found;
	}

	return newArray;
```

Reply:
I think this might be a bit *"slower"* in some edgcases but it's way more readable :+1:
Thanks :heart_eyes: